src/flamegraph/svg.rs

```diff
@@ -48,224 +48,270 @@ pub(super) struct StyleOptions<'a> {
-        key: b"y",
-        value: Cow::from(buf[y].as_bytes()),
-    });
-    text.push_attribute(Attribute {
-        key: b"font-size",
-        value: Cow::from(buf[fs].as_bytes()),
-    });
-    text.push_attribute(Attribute::from(("font-family", "Verdana")));
-    text.push_attribute(Attribute::from(("fill", item.color)));
-    svg.write_event(Event::Start(text))?;
-    let s = match item.text {
-        TextArgument::String(ref s) => &*s,
-        TextArgument::FromBuffer(i) => &buf[i],
-    };
-    svg.write_event(Event::Text(BytesText::from_plain_str(s)))?;
-    svg.write_event(Event::End(BytesEnd::borrowed(b"text")))?;
-    Ok(())
+        self.write_str(
+            svg,
+            &mut buf,
+            TextItem {
+                color: "rgb(0, 0, 0)",
+                size: self.font_size,
+                x: (self.image_width - super::XPAD - 100) as f64,
+                y: (self.font_size * 2) as f64,
+                text: "Search".into(),
+                location: None,
+                extra: vec![
+                    ("id", "search"),
+                    ("onmouseover", "searchover()"),
+                    ("onmouseout", "searchout()"),
+                    ("onclick", "search_prompt()"),
+                    ("style", "opacity:0.1;cursor:pointer"),
+                ],
+            },
+        )?;
+
+        self.write_str(
+            svg,
+            &mut buf,
+            TextItem {
+                color: "rgb(0, 0, 0)",
+                size: self.font_size,
+                x: (self.image_width - super::XPAD - 100) as f64,
+                y: (style_options.imageheight - (self.ypad2() / 2)) as f64,
+                text: " ".into(),
+                location: None,
+                extra: iter::once(("id", "matched")),
+            },
+        )?;
+
+        Ok(())
+    }
+
+    pub(super) fn write_str<'a, W, I>(
+        self: &SVGWriter,
+        svg: &mut Writer<W>,
+        buf: &mut StrStack,
+        item: TextItem<'a, I>,
+    ) -> quick_xml::Result<()>
+    where
+        W: Write,
+        I: IntoIterator<Item = (&'a str, &'a str)>,
+    {
+        let x = write!(buf, "{:.2}", item.x);
+        let y = write!(buf, "{:.2}", item.y);
+        let fs = write!(buf, "{}", item.size);
+        let mut text = BytesStart::borrowed_name(b"text").with_attributes(item.extra);
+        text.push_attribute(Attribute::from((
+            "text-anchor",
+            item.location.unwrap_or("left"),
+        )));
+        text.push_attribute(Attribute {
+            key: b"x",
+            value: Cow::from(buf[x].as_bytes()),
+        });
+        text.push_attribute(Attribute {
+            key: b"y",
+            value: Cow::from(buf[y].as_bytes()),
+        });
+        text.push_attribute(Attribute {
+            key: b"font-size",
+            value: Cow::from(buf[fs].as_bytes()),
+        });
+        text.push_attribute(Attribute::from(("font-family", self.font_type.as_str())));
```

Comment:
```suggestion
        text.push_attribute(Attribute::from(("font-family", &self.font_type)));
```

Reply:
```
   --> src/flamegraph/svg.rs:301:29
    |
301 |         text.push_attribute(Attribute::from(("font-family", &self.font_type)));
    |                             ^^^^^^^^^^^^^^^ the trait `std::convert::From<(&str, &std::string::String)>` is not implemented for `quick_xml::events::attributes::Attribute<'_>`
    |
    = help: the following implementations were found:
              <quick_xml::events::attributes::Attribute<'a> as std::convert::From<(&'a [u8], &'a [u8])>>
              <quick_xml::events::attributes::Attribute<'a> as std::convert::From<(&'a str, &'a str)>>
    = note: required by `std::convert::From::from`
```

Reply:
Ah, sorry, you need `&*self.font_type`. At that point it's also fine the way it was.